pyqtgraph/graphicsItems/ImageItem.py

```diff
@@ -299,8 +302,11 @@ def setImage(self, image=None, autoLevels=None, **kargs):
                 autoLevels = True
         if autoLevels:
             img = self.image
-            while img.size > 2**16:
+            level_samples = kargs.pop('levelSamples', 2**16) # use specified value or default of 2**16, remove from kargs
+            if level_samples < 2: level_samples = 2 # avoid endless loop, and keep at least two values for min, max
+            while img.size > level_samples:
                 img = img[::2, ::2]
+            print( img.size )
```

Comment:
We probably won't want this print statement in the merged code-base.

Reply:
resolved at @NilsNemitz 's request

Reply:
Ooops, my bad. Thank you for taking that out!